newdle/client/src/components/ParticipantTable.js

```diff
@@ -102,32 +109,45 @@ function AvailabilityRow({
             )}
           </div>
         </div>
+        {active && children}
       </Table.Cell>
-      <Table.Cell width={1} textAlign="right">
-        <Radio name="slot-id" value={startDt} checked={active} />
-      </Table.Cell>
+      {!finalized && (
+        <Table.Cell width={1} textAlign="right">
+          <Radio name="slot-id" value={startDt} checked={active} />
+        </Table.Cell>
+      )}
     </Table.Row>
   );
 }
 
-export default function ParticipantTable({finalDate, setFinalDate}) {
+export default function ParticipantTable({finalDate, setFinalDate, finalized, children}) {
   const availabilityData = useSelector(getParticipantAvailability);
 
   if (availabilityData.length === 0) {
     return null;
   }
 
+  if (finalized) {
+    availabilityData.some(
```

Comment:
`some()` with side effects is a bit strange... I think creating a new array with the correct order would be better.

Reply:
Then we would have to use conditions to know which array to use (whether it's sorted version or not)

Reply:
You can check inside `getParticipantAvailability` if a slot is the final one using the `final_dt` from the `getNewdle` selector and then sort the list accordingly